index.js

```diff
@@ -698,11 +710,17 @@ async function fetchRepoExtensions(contents_url, branch, stars) {
  */
 async function fetchThemes(contents_url, branch, stars) {
     try {
+        let manifests;
         const regex_result = contents_url.match(/https:\/\/api\.github\.com\/repos\/(?<user>.+)\/(?<repo>.+)\/contents/);
         // TODO: err handling?
         if (!regex_result || !regex_result.groups) return null;
         let { user, repo } = regex_result.groups;
-        let manifests = await getRepoManifest(user, repo, branch);
+        if (window.sessionStorage.getItem(user + "-" + repo)) {
+            manifests = JSON.parse(window.sessionStorage.getItem(user + "-" + repo));
+        } else {
+            manifests = await getRepoManifest(user, repo, branch);
```

Comment:
This is the same thing the `extension.js` is caching, I think? So we could cache the results here as well?